### ISSUE TITLE
bent Laue crystal with both sagittal bending and meridional bending

### ISSUE THREAD

**Issue by Zeelyne** [kklmn/xrt#127]
Hello，
I am recently working on simulating a bent Laue crystal with both sagittal bending and meridional bending. So far, I haven’t found such optic element in the _oes.py_ and _oes bases.py_. So, I am thinking about constructing one in the _oes.py_, based on the existing optic elements. According to you, should I construct this element, based on the _DCMwithSagittalFocusing_ class, or on the _BentLaueCylinder_ class? 
Thank you very much for your indication and have a nice day!

Best regards,
Ziling

**Comment by kklmn** [comment]
Hi,

I would use `GeneralBraggToroid` class. The method `local_n()` returns a 6-item list, the first 3 arrays being the coordinates of atomic plane normals and the last 3 arrays being the coordinates of surface normals. You need to change the first 3. Multiply them in a vector product to [1, 0, 0] to get them along the surface.

**Comment by Zeelyne** [comment]
Hello,
Great, thank you very much for you suggestion! I will try this.
Best regards,
Zeelyne

**Comment by Zeelyne** [comment]
Hello,
Great, thank you very much for you suggestion! I will try this.
Best regards,
Zeelyne

**Comment by kklmn** [comment]
2. I guess you yourself know where the atomic planes should be directed to. So position them accordingly. Most typically, the normal in Laue case is at 90 deg to the normal in Bragg case, so I suggested rotating it around [1, 0, 0] by 90 deg, which can be achieved by a *vector* product (aka cross product). But you can do it in any easiest way for you.

**Comment by Zeelyne** [comment]
Hello dear developer,
Thank you very much for your suggestion. I have tried your advice by modifying the oes.py as the followings, like the figure below:
![generalcode](https://user-images.githubusercontent.com/113894773/205854802-5b53510e-4005-439f-b491-21a22828c7cc.png)

After modification of the oes.py, I have encountered some other questions:
1.  As indicated by the following pictures, I have to set crystalSi01：geom=r“Bragg" in order to get the rays passing through the crystal via a Laue reflected mode. Is this normal?
![LAue](https://user-images.githubusercontent.com/113894773/205855296-ffe292c5-abda-4e68-991a-96026e47d063.png)
![BRagg](https://user-images.githubusercontent.com/113894773/205855306-9e6176a3-1603-44d6-96ad-3a304c0972da.png)

2.If the pitch angle value is carefully set, we can have the yellow arrow of [111] bragg plane intersecting the angle between the incident rays and the exit rays in XRTGLOW. But it is not always the case if the pitch angle is randomly set. Is this normal?

4.When modifying the Rm and Rs of the bent crystal (roes.GeneralBraggToroid), in XRTGLOW, the thickness of the crystal seems inhomogeneous: it is thicker at the edges of the crystal than at the center. Is it just a rendering effect?

![bent](https://user-images.githubusercontent.com/113894773/205855574-cd4f0ec0-b2c5-42cd-82fc-e3348e1a48b9.png)


Thank you so much, dear developer, for your continuous help and guidance.
Best regards,
Zeelyne
